src/Internal/ProductAttributesLookup/DataRegenerator.php

```diff
@@ -345,11 +338,19 @@ private function add_initiate_regeneration_entry_to_tools_array( array $tools_ar
 	 * @throws \Exception The regeneration is already in progress.
 	 */
 	private function initiate_regeneration_from_tools_page() {
-		if ( $this->data_store->regeneration_is_in_progress() ) {
-			throw new \Exception( 'Product attributes lookup table is already regenerating.' );
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+		if ( ! isset( $_REQUEST['_wpnonce'] ) || false === wp_verify_nonce( $_REQUEST['_wpnonce'], 'debug_action' ) ) {
+			throw new \Exception( 'Invalid nonce' );
 		}
 
-		$this->initiate_regeneration();
+		if ( isset( $_REQUEST['regenerate_product_attribute_lookup_data_product_id'] ) ) {
+			$product_id = (int) $_REQUEST['regenerate_product_attribute_lookup_data_product_id'];
+			$this->check_can_do_lookup_table_regeneration( $product_id );
+			$this->data_store->create_data_for_product( $product_id );
+		} else {
+			$this->check_can_do_lookup_table_regeneration();
```

Comment:
Small nitpick here but we probably can remove the `check` from the function name because `can` already means to check.